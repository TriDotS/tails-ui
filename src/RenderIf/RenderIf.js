import PropTypes from 'prop-types'

const RenderIf = ({ children, condition }) => (condition ? children : null)

RenderIf.propTypes = {
  children: PropTypes.node,
  condition: PropTypes.bool
}

export default RenderIf
