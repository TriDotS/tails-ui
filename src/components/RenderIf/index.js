import PropTypes from 'prop-types'

const RenderIf = ({ condition, children }) => (condition ? children : null)

RenderIf.propTypes = {
  condition: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
}

export default RenderIf
