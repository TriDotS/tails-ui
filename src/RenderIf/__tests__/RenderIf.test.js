import React from 'react'
import RenderIf from '../RenderIf'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<RenderIf condition={1 === 1}>Will this render?</RenderIf>, div)
})

test('render if snapshot', () => {
  const tree = renderer
    .create(<RenderIf condition={1 === 1}>Will this render?</RenderIf>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
