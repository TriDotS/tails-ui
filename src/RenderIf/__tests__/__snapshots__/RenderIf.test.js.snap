// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render if snapshot 1`] = `"Will this render?"`;
